listeners.go

```diff
@@ -652,6 +714,11 @@ var _ quic.OOBCapablePacketConn = (*fakeClosePacketConn)(nil)
 // but doesn't actually use these methods, the only methods needed are `ReadMsgUDP` and `SyscallConn`.
 var _ net.Conn = (*fakeClosePacketConn)(nil)
 
+// Unwrap returns the underlying net.UDPConn for quic-go optimization
+func (fcpc *fakeClosePacketConn) Unwrap() any {
+	return fcpc.UDPConn
+}
```

Comment:
Is this an interface quic-go supports? Cool, I didn't know that.

Reply:
Na, this just exposes the underlying socket so quic-go can optimize it. In this case it's a `*net.UDPConn`.

Reply:
So quic-go needs to be patched to call this method?

Reply:
We call this method ourselves, by passing `*net.UDPConn` or `*net.UnixConn` to `quic.ListenEarly`. And since we're managing the connection using `quicListener`, `fakeClose` shouldn't apply to it anyway.